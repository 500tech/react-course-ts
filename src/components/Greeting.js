import React from 'react';
import styled from 'styled-components';

const GreetingTitle = styled.h1`
  color: violet;
`;

export default function Greeting({ name = 'stranger' }) {
  return <GreetingTitle>Hello, {name}!</GreetingTitle>;
}
