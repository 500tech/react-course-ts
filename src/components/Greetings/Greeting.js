import React from 'react'; // why do we import React?
import styled from 'styled-components';

const StyledTitle = styled.h1`
  color: violet;
`;

export const RightDiv = styled.div`
  float: right;
`;

export function Greeting({ name = 'stranger', count = 0 }) {
  return (
    <StyledTitle>
      <span>Hello, {name}!</span>
      <RightDiv>{count}</RightDiv>
    </StyledTitle>
  );
}
