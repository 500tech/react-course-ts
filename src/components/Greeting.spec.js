import React from 'react';
import { shallow } from 'enzyme';
import Greeting from './Greeting';
import toJson from 'enzyme-to-json';

describe('<Greeting />', () => {
  it('has a default greeting', () => {
    const wrapper = shallow(<Greeting />);
    expect(wrapper.get(0).props.children.join('')).toBe('Hello, stranger!');
  });
  it('greets user by name', () => {
    const wrapper = shallow(<Greeting name="Foo" />);
    expect(wrapper.get(0).props.children.join('')).toBe('Hello, Foo!');
  });
  it('is sane', () => {
    const wrapper = shallow(<Greeting />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
