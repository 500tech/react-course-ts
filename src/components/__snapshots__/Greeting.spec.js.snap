// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting /> is sane 1`] = `
<styled.h1>
  Hello, 
  stranger
  !
</styled.h1>
`;
